lib/l1_sender/src/batcher_model.rs

```diff
@@ -41,14 +41,10 @@ pub struct BatchMetadata {
 
 impl BatchMetadata {
     /// Gets batch metadata verification key hash.
-    ///
-    /// NOTE: Panics if the execution version is unsupported, which *should* never happen in practice.
-    /// We could propagate an error here, but then we need to change APIs everywhere.
-    /// Given it is unlikely to happen in practice, we opt for simplicity.
-    pub fn verification_key_hash(&self) -> &'static str {
-        ExecutionVersion::try_from(self.execution_version)
-            .expect("Unsupported execution version")
-            .vk_hash()
+    pub fn verification_key_hash(&self) -> anyhow::Result<&'static str> {
```

Comment:
I would've preferred this as actual value (`&'static str`), rather than `Result`. It only makes API more verbose for situations where it shouldn't. (I can't imagine setups where the batch won't have verification_key, unless we broke the release and didn't make the proper updates to keys and mappings. And if we did, we should panic earlier -- i.e. when we create a batch, rather than panicing in the prover API after we tried to get batches proven)